### PR TITLE
New major version without Xerces

### PR DESCRIPTION
**Pull request by kuniss**
This PR is intended to implement the new major version 4 of javapos-config-loader incorporating the following major changes
- replacing the Xerces XML parser implementation by a Javax based one (contribution by @mjpcger )
- a lot of source code improvements 
  - generic types, code smell removals, 
  - modernization of the 1999 code base, 
  - internal data structure hiding
- removal of deprecated methods and classes
- solving long standing issue #1 to enable javapos-config-loader for web-start environments

### CODE REVIEW

**Comment by kuniss** [comment]
@mjpcger, do you want to review this pull request before I'm merging it? Most of the work is from your contribution...

I would, however, publish a pre-release of the original branch to Maven Central such that it can be verified in local build environments.

File: src/main/java/jpos/config/JposConfigException.java
Comment (mjpcger) [inline src/main/java/jpos/config/JposConfigException.java]: JavaDoc comment missing

File: src/main/java/jpos/config/JposEntryRegistryEvent.java
Comment (mjpcger) [inline src/main/java/jpos/config/JposEntryRegistryEvent.java]: Meaningful JavaDoc comment missing

File: src/main/java/jpos/loader/JposLoaderException.java
Comment (mjpcger) [inline src/main/java/jpos/loader/JposLoaderException.java]: Meaningful JavaDoc comment missing

File: src/main/java/jpos/profile/ProfileException.java
Comment (mjpcger) [inline src/main/java/jpos/profile/ProfileException.java]: Meaningful JavaDoc comment missing

**Review by mjpcger — COMMENTED** [review]
Se.ems to be good, only some JavaDoc comments should not remain empty

File: src/main/java/jpos/config/JposConfigException.java
Comment (kuniss) [reply src/main/java/jpos/config/JposConfigException.java]: I removed the empty comment. It had been added automatically by Eclipse when generating the serial UID to solve a compiler warning.

File: src/main/java/jpos/config/JposEntryRegistryEvent.java
Comment (kuniss) [reply src/main/java/jpos/config/JposEntryRegistryEvent.java]: I removed the empty comment. It had been added automatically by Eclipse when generating the serial UID to solve a compiler warning.

File: src/main/java/jpos/loader/JposLoaderException.java
Comment (kuniss) [reply src/main/java/jpos/loader/JposLoaderException.java]: I removed the empty comment. It had been added automatically by Eclipse when generating the serial UID to solve a compiler warning.

File: src/main/java/jpos/profile/ProfileException.java
Comment (kuniss) [reply src/main/java/jpos/profile/ProfileException.java]: I removed the empty comment. It had been added automatically by Eclipse when generating the serial UID to solve a compiler warning.

**Comment by kuniss** [comment]
@mjpcger could you please make a final review and approve the PR, please?
You should have something like the following in the GitHub GUI under the _Review Changes_ button:
image

I could merge without your approval (because I'm the project admin 😄), however, I would like to try the "clean" process way...
Thanks!

**Comment by kuniss** [comment]
A RC built from this PR has been published to Maven Central's snapshot repository at https://oss.sonatype.org/content/repositories/snapshots/org/javapos/javapos-config-loader/4.0.0-SNAPSHOT/. 

It may be downloaded from there or resolved by dependency resolution if configured appropriately, e.g. for Gradle:
```
repositories {
    maven {
        url 'https://oss.sonatype.org/content/repositories/snapshots'
    }
}

dependencies {
    implementation 'org.javapos:javapos-config-loader:4.0.0+'
}
```

Who ever is interested, please integrate in your environment for testing purposes and report back any issue here in this PR.

**Review by mjpcger — APPROVED** [review]
Looks good for me, no change necessary.

**Comment by mjpcger** [comment]
More approval is not possible for me. Looks as if I have no write access and therefore my approval is not enough to change the **Review required** status.

**Comment by kuniss** [comment]
Thanks, @mjpcger!
I adjusted the  branch rules. Now I can merge according to the rules. 😄

**Comment by kuniss** [comment]
Who ever is interested: We just encountered in our off-side integration test that some of our JavaPOS XML configuration files are not read in correctly by the new javapos-config-loader-4.0.0-RC1, Those were read in without errors by the Xerces implementation, however.

We are investigating...

**Comment by kuniss** [comment]
The reported regression was a misinterpretation. In fact, it new XML populator is just more accurate. If the DTD is given in the JavaPOS XML configuration file like 
´´´
<!DOCTYPE JposEntries PUBLIC "-//JavaPOS//DTD//EN"
                             "jpos/res/jcl.dtd">
´´´
then file _jpos/res/jcl.dtd_ must be in the classpath, otherwise the validation fails and no entries are read in.

**Comment by kuniss** [comment]
I improved the tracing a bit. So, I'm going to publish a new SNAPSHOT on Maven Central's snapshot repository - RC2.

**Pull request merged**
